### PR TITLE
Lbry

### LINKED ISSUE #321: Bitchute and others
Since yt is protecting us from "harmful content" would it be possible to include other from YouTube-dl supported sites?

### PR DESCRIPTION
**Pull request by Euro20179**
Fixes #321
credit to @trueauracoral for helping with the api.
* also fixes a bug with -P, where the text wouldn't display with thumbnails.

### CODE REVIEW

**Comment by odnar-dev** [comment]
does this work??

**Comment by Euro20179** [comment]
It should, i've done testing, i've put how to test it in #321.

File: ytfzf [hunk 1]
```diff
@@ -802,13 +804,46 @@ scrape_pt () {
...
      videos_data="$(get_video_data "$videos_json")"
      function_exists "on_video_data_gotten" && on_video_data_gotten "$videos_data" "$videos_json" "$pt_json"
      [ $show_thumbnails -eq 1 ] && download_thumbnails "$videos_json"
      unset pt_json
 }
 
+scrape_odysee () {
+    odysee_json=$(
+	curl \
+	    -s "https://lighthouse.lbry.com/search" \
```
Comment (odnar-dev) [inline ytfzf:817]: add this to show only videos
`-G --data-urlencode "mediaType=video,audio"`
add this to show more results
`-G --data-urlencode "size=30"`

File: ytfzf [hunk 2]
```diff
@@ -802,13 +804,46 @@ scrape_pt () {
...
+	    -G --data-urlencode "s=$*" \
+	    -G --data-urlencode "include=channel,title,thumbnail_url,duration,cq_created_at,content_type,description"
+    )
+    [ -z "$odysee_json" ] && print_error "ERROR[#01]: Couldn't curl website. Please check your network and try again.\n" && exit 1
+
+    videos_json=$(printf "%s" "$odysee_json" |\
+	jq '[ .[] | select(.content_type == "video/mp4" or .content_type == "video/m4v" or .content_type == "video/webm") |
```
Comment (odnar-dev) [inline ytfzf:824]: remove this not needed anymore
`| select(.content_type == "video/mp4" or .content_type == "video/m4v" or .content_type == "video/webm")`

**Comment by Euro20179** [comment]
Is there like some place for lighthouse documentation cause I couldn't find it?

**Comment by odnar-dev** [comment]
> Is there like some place for lighthouse documentation cause I couldn't find it?
me neither , i used the network monitor tool in firefox to find how the api work

**Comment by trueauracoral** [comment]
> Is there like some place for lighthouse documentation cause I couldn't find it?

I also couldn't find anything on it. I just emailed the developer. https://github.com/lbryio/lighthouse At the bottom there is a contact page. So I used that. He is very active. If you have any questions you could definitely redirect it his way. 

Backstory to how I made something/found the API.
I found lighthouse api randomly when trying to find make terminal odysee search engine. It was kind of weird to find. Then I coded something. After a lot of errors. I made it work. Then found this awesome project. Went to the issues tab, found the Bitchute and others issue. And proposed my project.

**Comment by Euro20179** [comment]
> i used the network monitor tool in firefox to find how the api work

I am kind of curious as to how you did this, but nice.

> At the bottom there is a contact page. So I used that. He is very active. If you have any questions you could definitely redirect it his way.

This is good to know, thank you.

**Comment by odnar-dev** [comment]
> I am kind of curious as to how you did this, but nice.

i just opened [https://odysee.com/$/search?q=python](https://odysee.com/$/search?q=python) in browser, played with filters 
image
while monitoring all outgoing requests 
image

**Comment by Euro20179** [comment]
ohh, didn't realize odysee used it.

**Comment by trueauracoral** [comment]
Is there anyway I can set up and use the lighthouse commit?

> ohh, didn't realize odysee used it.

wowsers the power of open source.

**Comment by Euro20179** [comment]
> is there anyway I can set up and use the lighthouse commit?

If you're talking about my pull request, my last comment in #321 explains how to use it. Otherwise I'm not sure what you mean.

**Comment by trueauracoral** [comment]
> If you're talking about my pull request, my last comment in #321 explains how to use it. Otherwise I'm not sure what you mean.

Thanks that was what I needed.

File: ytfzf [hunk 3]
```diff
@@ -802,13 +829,60 @@ scrape_pt () {
...
 
+scrape_odysee () {
+    odysee_json=$(
+	curl \
+	    -s "https://lighthouse.lbry.com/search" \
+	    -G --data-urlencode "s=$*" \
+	    -G --data-urlencode "mediaType=video,audio" \
+	    -G --data-urlencode "include=channel,title,thumbnail_url,duration,cq_created_at,description" \
+	    -G --data-urlencode "size=$odysee_video_search_count" \
+	    -G --data-urlencode "time_filter="$(printf "$upload_date_filter" | tr -d '-')"" \
```
Comment (pystardust) [inline ytfzf:848]: curl `-G` indicates a get request, it suffices to put it once.

The rest seems great

**Review by pystardust — APPROVED** [review]

**Pull request merged**